scripts/delete-old-app-engine-versions.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+#
+# Deletes old Google App Engine deployments of Terra UI in an environment. You
+# MUST have GNU date installed to be able to use this script.
+#
+# USAGE: ./delete-old-app-engine-versions.sh ENV
+#   ENV must be one of dev, alpha, staging, or perf
+#
+
+set -eu
+set -o pipefail
+
+# check if colors are supported in the terminal
+check_color_support() {
+    NCOLORS=$(tput colors)
+    if [ "$NCOLORS" -ge 8 ]; then
+        BLD="$(tput bold)"
+        RED="$(tput setaf 1)"
+        GRN="$(tput setaf 2)"
+        RST="$(tput sgr0)"
+    else
+        BLD=""
+        RED=""
+        GRN=""
+        RST=""
+    fi
+    INFO="${BLD}+${RST}"
+}
+
+# print out usage to stdout
+usage() {
+    printf "Usage: %s ${BLD}ENV${RST}\n  ${BLD}ENV${RST} must be one of dev, alpha, staging, or perf\n" "$0"
+    exit 0
+}
+
+# print out error with help message to stderr and exit
+error() {
+    printf "${RED}ERROR: %s${RST}\n\nTry ${BLD}%s --help${RST} to see a list of all options.\n" "$1" "$0" >&2
+    exit 1
+}
+
+# print out error to stderr and exit
+abort() {
+    printf "${RED}ABORT: %s${RST}\n" "$1" >&2
+    exit 1
+}
+
+# always reset gcloud project regardless of error or exit
+reset() {
+    gcloud config set project "${OLD_PROJECT}" 2>/dev/null
+}
+
+# ensure that date from GNU coreutils is installed
+check_gnu_date_installed() {
+    if ! date --version 1>/dev/null 2>&1; then
+        abort "date from GNU coreutils is required"
+    fi
+}
+
+# switch gcloud project to appropriate terra ui environment
+switch_gcloud_project() {
+    gcloud config set project "${NEW_PROJECT}" 2>/dev/null
```

Comment:
Instead of modifying `gcloud` configuration, would it be preferable to add a `--project` flag to all invocations of `gcloud`?

Reply:
I was thinking the same.

Reply:
I'll give that a shot 👍